testing/go/alter_table_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package _go
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestAlterTableAddPrimaryKey(t *testing.T) {
+	RunScripts(t, []ScriptTest{
+		{
+			Name: "Add Primary Key",
+			SetUpScript: []string{
+				"CREATE TABLE test1 (a INT, b INT);",
+				"CREATE TABLE test2 (a INT, b INT, c INT);",
+				"CREATE TABLE pkTable1 (a INT PRIMARY KEY);",
+			},
+			Assertions: []ScriptTestAssertion{
+				{
```

Comment:
We should add an assertion for when the table already has duplicate data that would fail the pk creation